sw/device/tests/i2c_host_clock_stretching_test.c

```diff
@@ -0,0 +1,255 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+#include <assert.h>
+
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/base/macros.h"
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_i2c.h"
+#include "sw/device/lib/dif/dif_rv_plic.h"
+#include "sw/device/lib/runtime/hart.h"
+#include "sw/device/lib/runtime/irq.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/runtime/print.h"
+#include "sw/device/lib/testing/i2c_testutils.h"
+#include "sw/device/lib/testing/rv_core_ibex_testutils.h"
+#include "sw/device/lib/testing/rv_plic_testutils.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "i2c_regs.h"  // Generated.
+
+static_assert(__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__,
+              "This test assumes the target platform is little endian.");
+
+OTTF_DEFINE_TEST_CONFIG();
+
+enum {
+
+  kHart = kTopEarlgreyPlicTargetIbex0,
+
+  kDeviceAddr = 0x22,
+  // Registers
+  kSwResetReg = 0xDF,
+  kManufacturerIdReg = 0xDE,
+  kProductIdReg = 0xDD,
+  kRxDelayReg = 0xDC,
+  kTxDelayReg = 0xDB,
+  kCache63BitsReg = 0xCF,
+  // Registers values
+  kManufacturerId = 0xA1,
+  kProductId = 0xA2,
+};
+
+/**
+ * Declared volatile because it is referenced in the main program flow as well
+ * as the ISR.
+ */
+// Hold the test result.
+static volatile status_t test_result;
```

Comment:
This should be a dedicated word `irq_result` as all of the usages of `test_result` in `test_main` should be thought of as read-modify-write cycles.  You could end up with either the irq result or the test_main results clobbering each other.

Reply:
Fixed.